File: images/toolchain/debian-trixie/Dockerfile.scala
# Toolchain image: Scala.
#
# scala-cli is the modern launcher (it can run scripts, manage a project, and
# drive Bloop); sbt remains the incumbent build tool for large or legacy
# projects. Both are shipped so either workflow works. Extends the java
# toolchain for the JDK, Gradle and the cacerts trust the egress CA lands in.
ARG BASE_IMAGE=toolchain-java:debian-trixie
FROM ${BASE_IMAGE}
COPY cache/scala-cli-x86_64-pc-linux.gz /tmp/scala-cli.gz
RUN gunzip -c /tmp/scala-cli.gz > /usr/local/bin/scala-cli && chmod +x /usr/local/bin/scala-cli && rm /tmp/scala-cli.gz
COPY cache/sbt-1.13.0.tgz /tmp/sbt.tgz
RUN tar -xzf /tmp/sbt.tgz -C /opt && rm /tmp/sbt.tgz
ENV PATH=/opt/sbt/bin:$PATH
RUN scala-cli version && sbt --script-version

File: images/toolchain/debian-trixie/Dockerfile.zig
# Toolchain image: Zig.
#
# Zig has no central package registry: `build.zig.zon` fetches dependencies by
# URL, so there is nothing to cache with a protocol adapter. The compiler is a
# self-contained tarball.
ARG BASE_IMAGE=toolchain-base:debian-trixie
FROM ${BASE_IMAGE}
COPY cache/zig-x86_64-linux-0.16.0.tar.xz /tmp/zig.tar.xz
RUN mkdir -p /opt/zig && tar -xJf /tmp/zig.tar.xz -C /opt/zig --strip-components=1 && rm /tmp/zig.tar.xz
ENV PATH=/opt/zig:$PATH
RUN zig version
